projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.service.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@angular/core';
+import { combineLatest, EMPTY, Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { CmsMerchandisingCarouselComponent } from '../../../cds-models/cms.model';
+import { CdsMerchandisingProductService } from '../../facade/cds-merchandising-product.service';
+import {
+  MerchandisingProduct,
+  MerchandisingProducts,
+} from '../../model/merchandising-products.model';
+import { MerchandisingCarouselModel } from './merchandising-carousel.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MerchandisingCarouselComponentService {
+  constructor(
+    protected cdsMerchandisingProductService: CdsMerchandisingProductService
+  ) {}
+
+  getMerchandisingCarouselModel(
+    cmsComponent: CmsMerchandisingCarouselComponent
+  ): Observable<MerchandisingCarouselModel> {
+    return combineLatest([
+      of(cmsComponent),
```

Comment:
Looking into if we should inject or not with Nikola

Reply:
This way it's done currently is fine.